src/parser_api.jl

```diff
@@ -111,56 +54,37 @@ function parse(stream::ParseStream; rule::Symbol=:toplevel)
     stream
 end
 
-function parse(::Type{T}, io::IO;
-               rule::Symbol=:toplevel, version=VERSION, kws...) where {T}
+"""
+    parse!(io::IO; rule=:toplevel, version=VERSION)
+
+Parse Julia source code from a seekable `IO` object. The output is a tuple
+`(tree, diagnostics)`. The `io` position will be set to the next byte of input
+after parsing.
```

Comment:
"to the first unparsed byte of input" or something like that (is that even correct when the whole io content is consumed?)? "next byte" without any context could be kinda confusing.

Reply:
ah this is so hard to word in a way which is clear and correct. How about

"When `parse!` returns, the stream `io` is positioned directly after the last byte which was consumed during parsing."